samples/nrf_desktop/src/hw_interface/battery_discrete.c

```diff
@@ -392,6 +419,33 @@ static bool event_handler(const struct event_header *eh)
 		return sampling;
 	}
 
+	if (is_usb_state_event(eh)) {
+		struct usb_state_event *event = cast_usb_state_event(eh);
+		int err;
+
+		switch (event->state) {
+		case USB_STATE_SUSPENDED:
+		case USB_STATE_DISCONNECTED:
+			err = charging_disable();
+			break;
+
+		case USB_STATE_POWERED:
+		case USB_STATE_ACTIVE:
+			err = charging_enable();
```

Comment:
are we sure this will not happen before module is initialized?

Reply:
Both usb_state and battery_discrete are initialized on main module state ready, so it shouldn't be a problem